xarray/core/duck_array_ops.py

```diff
@@ -72,7 +72,7 @@ def fail_on_dask_array_input(values, msg=None, func_name=None):
 
 
 if hasattr(np, "isnat") and (
-    dask_array is None or hasattr(dask_array_type, "__array_ufunc__")
+    dask_array_type == () or hasattr(dask_array_type[0], "__array_ufunc__")
```

Comment:
I'm really not sure if this is the best way to make sure we actually call `hasattr` on `dask.array.Array`:
https://github.com/pydata/xarray/blob/90e734a55792f3c19e795df110fc3501c609d191/xarray/core/pycompat.py#L5-L11